src/prng.rs

```diff
@@ -132,6 +132,28 @@ where
     }
 }
 
+impl<F> CoinToss for F
+where
+    F: FieldElement,
+{
+    fn sample<S>(seed_stream: &mut S) -> Self
+    where
+        S: SeedStream,
+    {
+        // This is analogous to `Prng::get()`, but does not make use of a persistent buffer of
+        // `SeedStream` output.
+        let mut buffer = vec![0u8; F::ENCODED_SIZE];
```

Comment:
It's a shame we can't (yet) use an array here. I wonder if we should start leaving TODO comments so that once the necessary compiler feature lands, we can save some allocations in places like this.

Reply:
Actually, last time this came up in `decode_fieldvec()`, we made a fixed size 64-byte array on the stack. We could do the same here, I'm optimistic that it would be an improvement.

Reply:
Nice, this is another across-the-board benchmark win.